### PR TITLE
[4][com_categories] Verify that the alias is unique before batch move

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #39137

### Summary of Changes
 Verify that the alias is unique before batch move

### Testing Instructions
see #39137

### Actual result BEFORE applying this Pull Request
categories moved even with same alias 

### Expected result AFTER applying this Pull Request
not moved cause of same alias

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on ffe8ed72c99ee86184a8db0b304e8b2ffa106cef<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39158">issues.joomla.org/tracker/joomla-cms/39158</a>.</sub>

File: administrator/components/com_categories/src/Model/CategoryModel.php [hunk 1]
```diff
@@ -1163,6 +1163,19 @@ protected function batchMove($value, $pks, $contexts)
 
                     return false;
                 }
+     
+                // Verify that the alias is unique before move
+                $conditions = [
+                    'alias'     => $this->table->alias, 
+                    'parent_id' => $parentId, 
+                    'extension' => $extension
```
Comment (sandewt) [inline administrator/components/com_categories/src/Model/CategoryModel.php]: End array with a comma:
`extension' => $extension,`

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on ffe8ed72c99ee86184a8db0b304e8b2ffa106cef<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39158">issues.joomla.org/tracker/joomla-cms/39158</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on a67e36745e9c64fd37cd7e3fc296a6c94f2fb3f9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39158">issues.joomla.org/tracker/joomla-cms/39158</a>.</sub>

**Comment by ChristineWk** [comment]
Before PR:

screen shot 2022-11-08 at 22 02 58

With Patch:

screen shot 2022-11-08 at 22 03 43<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39158">issues.joomla.org/tracker/joomla-cms/39158</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39158">issues.joomla.org/tracker/joomla-cms/39158</a>.</sub>

**Comment by richard67** [comment]
Thanks all.

**Comment by wilsonge** [comment]
Sorry I missed this. We should be careful here. Adding the check is good but as we're batch moving and may already have stored previous batch categories if we're doing multiple we need to specify which category failed in the error. We should also probably note some categories may have successfully completed batch (we can use a Boolean to detect this). Potentially we could use continue for the for loop rather than returning false so we continue onto other items and list the ones that already exist

**Comment by richard67** [comment]
@wilsonge Shall we revert it?

**Comment by apiviuw** [comment]
verifikasi

**Pull request merged**